libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -152,6 +152,39 @@ string YulUtilFunctions::storeLiteralInMemoryFunction(string const& _literal)
 	});
 }
 
+string YulUtilFunctions::copyLiteralToStorageFunction(string const& _literal)
+{
+	string functionName = "copy_literal_to_storage_" + util::toHex(util::keccak256(_literal).asBytes());
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		size_t words = (_literal.length() + 31) / 32;
+		vector<map<string, string>> wordParams(words);
+		for (size_t i = 0; i < words; ++i)
+		{
+			wordParams[i]["offset"] = to_string(i);
+			wordParams[i]["wordValue"] = formatAsStringOrNumber(_literal.substr(32 * i, 32));
+		}
+
+		return Whiskers(R"(
+			function <functionName>(slot) {
+				<resizeArray>(slot, <length>)
+				let dstPtr := <?long><dataLocation>(slot)<!long>slot</long>
+				<#word>
+					sstore(add(dstPtr, <offset>), <?long><wordValue><!long>add(<wordValue>, mul(2, <length>))</long>)
+				</word>
+				<?long>sstore(slot, add(1, mul(2, <length>)))</long>
```

Comment:
Isn't the length already written to in the resize function? Maybe we could even go as far as to *not* calling the resize function, but just manually clearing the tail with ``clearStorageRange`` when decreasing the size here... but it's not nice... not quite sure, but I'm wondering.

Maybe using the resize function and just writing the length again in the "short" case is fine, but the write in the long case we can definitely skip, can't we?

Reply:
We can skip writing length in case of long one, and I did it like that for now